develop/learning-paths/migration/articles/01-intro/01-from-a-monolith-to-a-modular-architecture.markdown

```diff
@@ -0,0 +1,124 @@
+easily in the development process, as we hadn't yet found an easy way to facilitate and enforce
+loose coupling. So we also prioritized finding a framework to help Liferay
+engineers and contributors develop loosely coupled components on Liferay.
+
+One last major goal related to both releasability and loose coupling was finding
+the best way to modularize Liferay's core, apps, and frameworks into independent
+lightweight components. We wanted to do away with the massive monolith and
+refactor it into components that would be easy to maintain and iterate on
+separately and frequently. In our research, we drew inspiration from the concept
+of microservices, which we'll discuss next. 
+
+## The Microservices Concept
+
+The goal of a microservices is to do one (and only one) thing very
+well. Each microservice is small and stripped down of all but what it requires
+to meet its purpose. Their independent too. You can independently deploy each
+microservice and use it. But in addition to being independent, each microservice
+is composable, meaning it can be used with other microservices to do more
+complex things. 
+
+![Figure 1: Liferay drew inspiration from the concept of microservices. Their size, self reliance, and composability, make them ideal for a modular architecture.](../../images/microservice.png)
+
+In essence, we've rebuilt Liferay as microservices, to gain the ability to
+deploy new versions of the services in a few weeks, or even days, as compared
+previously to months. We've encapsulated these microservices in OSGi modules,
+which are a standards based deployment structure for deploying to dynamic Java
+systems.
+
+All of Liferay 7's modules can be developed and iterated on independently. Each
+module can have it's own release cycle. Developers can innovate rapidly over
+their modules. As with their modules, they can focus on providing the single
+purpose of their module.
+
+We'll go into details of OSGi modules later. For now, take comfort that we've
+chipped away at the Liferay 6.2 monolith to bring you Liferay 7 as a robust
+system of modules. Next, we'll consider how their delivered to you as a Liferay
+consumer. 
+
+## Core, Apps, and Suites
+
+We've taken Liferay Portal the monolith, broken it down to its parts, decoupled
+them, and re-engineered them into a much more manageable and agile system. Before
+going into details of how we implemented particulars of modularity, let's take a
+look at how Liferay 7's modules are arranged.
+
+![Figure 2: Liferay 7 is composed of the Liferay Core and several suites, each with their own set of application and framework modules.](../../images/core-suites-apps.png)
+
+What does all this mean? Has Liferay Portal simply been repackaged? No;
+it's been redeveloped into independent robust modules that work together to
+bring you a full-fledged portal. And on top of that, we've made the modules
+accessible in aggregations that are easier to release. We'll brief you on them
+now. 
+
```

Comment:
There's a lot of focus on "we," and what "we've" done in this text. It should be refocused on "you." The idea is that you, as a Liferay developer, are upgrading your knowledge of Liferay.